### PR TITLE
Inject OapSparkSqlParser by SparkSessionExtension

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

Inject OapSparkSqlParser by SparkSessionExtension.
SparkSqlParser will be removed after #999 is merged.

About "SqlBase.g4", needs to be kept currently.
## How was this patch tested?

Existed unit tests.

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import org.apache.spark.sql.execution.command._
+import org.apache.spark.sql.execution.datasources._
+import org.apache.spark.sql.execution.datasources.oap.index._
+import org.apache.spark.sql.internal.{HiveSerDe, SQLConf, VariableSubstitution}
+import org.apache.spark.sql.types.StructType
+
+/**
+ * Concrete parser for Spark SQL statements.
+ */
+ class OapSparkSqlParser(conf: SQLConf) extends AbstractSqlParser {
```
Comment (LuciferYang) [inline src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: I found there are similar code in spark source: class `CatalystSqlParser` and object `CatalystSqlParser` and object `CatalystSqlParser` comments `For test-only`.
Is there any risk in use object `OapSparkSqlParser` ? Can we use class `OapSparkSqlParser` as following?
```
class OapSparkSqlParser extends AbstractSqlParser {

  lazy val conf: SQLConf = SparkSession.getActiveSession.get.sessionState.conf

  lazy val astBuilder = new OapSparkSqlAstBuilder(conf)

  private lazy val substitutor = new VariableSubstitution(conf)

  protected override def parse[T](command: String)(toResult: SqlBaseParser => T): T = {
    super.parse(substitutor.substitute(command))(toResult)
  }
}
``` and remove object `OapSparkSqlParser` , or use object `OapSparkSqlParser` and remove class `OapSparkSqlParser`, just one is enough.

File: src/main/scala/org/apache/spark/sql/OapExtensions.scala [hunk 2]
```diff
@@ -27,5 +28,6 @@ class OapExtensions extends (SparkSessionExtensions => Unit) {
     extensions.injectPlannerStrategy(_ => OapSemiJoinStrategy)
     extensions.injectPlannerStrategy(_ => OapGroupAggregateStrategy)
     extensions.injectPlannerStrategy(_ => OapFileSourceStrategy)
+    extensions.injectParser((_, _) => OapSparkSqlParser)
```
Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/OapExtensions.scala]: may need add a comments to line 31.

**Comment by LuciferYang** [comment]
Good work~

File: src/main/scala/org/apache/spark/sql/OapExtensions.scala (hunk 2 above)
Comment (zhixingheyi-tian) [reply src/main/scala/org/apache/spark/sql/OapExtensions.scala]: @LuciferYang 
I will add some comments here.

File: src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala (hunk 1 above)
Comment (zhixingheyi-tian) [reply src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: @LuciferYang 
Why add lazy here?

Comment (LuciferYang) [reply src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: no lazy will stack overflow

File: src/main/scala/org/apache/spark/sql/OapExtensions.scala (hunk 2 above)
Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/OapExtensions.scala]: @zhixingheyi-tian 
Is sessionState.conf eq  `new SQLConf` merge SparkContext.conf ? SparkContext.getOrCreate use a AtomicReference, it's not None after SparkContext create, `SparkSession.getActiveSession.get` is a InheritableThreadLocal, it maybe None, we can discuss this , I am not sure.

Comment (zhixingheyi-tian) [reply src/main/scala/org/apache/spark/sql/OapExtensions.scala]: @LuciferYang 
Hi, have you considered "SparkSession.getActiveSession.get.sessionState.conf" is feasible?

How  about “SparkSession.getDefaultSession.get.sessionState.conf”

File: src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
+/*
```
Comment (ConeyLiu) [inline src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: why does it under spark folder?

Comment (LuciferYang) [reply src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: @ConeyLiu really shouldn't @zhixingheyi-tian  I think we need a follow-up to move this class out.

Comment (LuciferYang) [reply src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: @ConeyLiu #1012 as a followup to move this file to oap folder.

Comment (ConeyLiu) [reply src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: @LuciferYang thanks for the work.

Comment (zhixingheyi-tian) [reply src/main/spark2.3.2/scala/org/apache/spark/sql/execution/OapSparkSqlParser.scala]: @LuciferYang @ConeyLiu 

Thanks for your kind reminder.

**Pull request merged**